### ISSUE TITLE
Problem with docker compose V2 support.

### ISSUE THREAD

**Issue by malylemireflo** [mariotoffia/FluentDocker#312]
Seems docker-compose command should be updated to docker compose in V2.
https://docs.docker.com/compose/migrate/
Base command should check for .AssumeComposeVersion(ComposeVersion.V2) and remove the hyphen.
I get some errors with the newest docker compose and the command with hypen.
See https://askubuntu.com/questions/1508129/docker-compose-giving-containerconfig-errors-after-update-today

**Comment by rsaugier** [comment]
Hello. I also get errors when using "docker-compose". More precisely this python error: `KeyError: 'ContainerConfig' `.
 It seems like [we're not the only ones](https://stackoverflow.com/questions/78380867/docker-compose-run-issue-2024-error-containerconfig) and some people on the net seem to have solved this error by using `docker compose` instead of `docker-compose`.
I suppose on some systems (say Windows with Docker Desktop) docker-compose can be an alias to docker compose (ie V2). But on others (say a Linux build agent in a popular commercial continuous integration service), it is possible that docker-compose points to V1 and docker compose is the only way to go to have V2.
So yes, fixing FluentDocker by using `docker compose` instead of `docker-compose` is probably the right thing to do.
I could try to take a look if nobody handles this, but not _immediately_.

**Comment by mariotoffia** [comment]
Hi @rsaugier  and @malylemireflo I need to take care of this. Put it on my TODO list and will fix it during my vacation this summer.

Cheers,
 Mario

**Comment by ben-wilson-mews** [comment]
@mariotoffia Just FYI github have deprecated docker-compose in the base action images as of July 29th (with a slow rollout). We were hit today. Currently looking into a solution

**Comment by ben-wilson-mews** [comment]
For anyone who needs a quick shim until support lands:

```
ln -s /usr/libexec/docker/cli-plugins/docker-compose /usr/local/bin/docker-compose
```

**Comment by discoaaron** [comment]
> @mariotoffia Just FYI github have deprecated docker-compose in the base action images as of July 29th (with a slow rollout). We were hit today. Currently looking into a solution

More info here
https://github.com/actions/runner-images/issues/9692

**Comment by RSR-SE** [comment]
> For anyone who needs a quick shim until support lands:
> 
> ```
> ln -s /usr/libexec/docker/cli-plugins/docker-compose /usr/local/bin/docker-compose
> ```

Thanks a lot @ben-wilson-mews this one saved my life :-)
I'm still too busy and the code of the lib looks quite complex so I lazily didn't attempt to fix this myself. In any case this is not a blocking point anymore for us thanks to this great symlink workaround.

**Comment by dosper7** [comment]
We are having this issue as well at my work. Is there any update?

**Comment by mariotoffia** [comment]
Yes, I've finally got some time to spend on this project... at last... I will implement the following:
* If no docker-compose is found -> revert to check if docker compose (subcommnad) do work - This should eliminate any explicit configuration in FluentAPI and thus "just work".

**Comment by mariotoffia** [comment]
> > [@mariotoffia](https://github.com/mariotoffia) Just FYI github have deprecated docker-compose in the base action images as of July 29th (with a slow rollout). We were hit today. Currently looking into a solution
> 
> More info here [actions/runner-images#9692](https://github.com/actions/runner-images/issues/9692)

Thanks, I saw that, I'm slowly starting to get time for this project so I will address this issue next

**Comment by jmira-dojo** [comment]
looks like it works 🎉 thank you @mariotoffia
